llmfoundry/utils/data_prep_utils.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import os
+import platform
+from glob import glob
+from typing import List, Optional
+
+import psutil
+from composer.utils import ObjectStore
+from torch.utils.data import DataLoader, Dataset
+
+
+def build_dataloader(dataset: Dataset,
+                     batch_size: int,
+                     num_workers: Optional[int] = None) -> DataLoader:
+    if num_workers is None:
+        # Multiple workers is only supported on linux machines
+        if 'linux' in platform.platform().lower():
```

Comment:
Can MAC run this code ? In that case, I would recommend do a `os.name == 'posix'` check. 

macOS and Linux are both POSIX-compliant. Hence, an application written for macOS should run on Linux.

Reply:
IIRC it only worked on linux specifically...but not certain.

Reply:
I'm actually going to remove this code altogether. It's from a refactor, but this refactor is now out of scope for this PR since the code no longer uses build_dataloader.